src/blueprints/tv2/timeline-object-factories/tv2-caspar-cg-timeline-object-factory.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  CasparCgMediaTimelineObject,
+  CasparCgTemplateTimelineObject,
+  CasparCgType
+} from '../../timeline-state-resolver-types/caspar-cg-types'
+import { Tv2CasparCgLayer, Tv2GraphicsLayer } from '../value-objects/tv2-layers'
+import { DeviceType } from '../../../model/enums/device-type'
+import { Tv2VideoClipManifestData } from '../value-objects/tv2-action-manifest-data'
+
+export class Tv2CasparCgTimelineObjectFactory {
```

Comment:
I agree with this, i've thought the same thing.

Reply:
Our current formatting conventions withs acronyms is to treat the acronym as a single "word" or "part of" a word. So for example in `TimelineObject`, `Timeline` and `Object` are both a "word" or "part of" a word. So an acronym would be the equivilant of either `Timeline` or `Object`.
By treaing acronyms like this, then when using camelCasing it becomes clearer that that string of characters forms an acronym.

In the case of CasparCG, `CG` is an acronym, so to follow our convention it should be written `Cg`, so CasparCG becomes `casparCg`.

CasparCG isn't the best example to showcase this, but take a look at "http post". If we follow your example we would have to write it like `HTTPPost`, but by camelCasing the acronym it becomes `httpPost`.
It becomes extremely clear when you have multiple acronyms in the same word. Just to take a random example: `HTTPRESTEndpoint` is not very easy to read, but `httpRestEndpoint` is.

Reply:
The concept timeline object is a term that consists of the two separate words _timeline_ and _object_.  Neither of these words are acronyms for a timeline object. `TO` would be the acronym.

`CG` in CasparCG might be an acronym, but`CasparCG` is a product name, and is a single word. The product name might have a specific casing assigned, to emphasize a certain pronunciation, but it is still a single word.

When we write variable names for things related to iNews, we don't write the variable as `iNewsThings`, but as `inewsThings`. The same should be true for CasparCG (`casparcgThings` instead of `casparCgThings`). From this I think that `Casparcg` will be the "right" formatting in class names.

If you have never heard of CasparCG, and encounter this, you would probably copy and paste it to a search engine, leading you to the official site, where you can see the the logo casing of the product name.

Reply:
I do write it as `iNewsThing`